packages/amplify-graphql-model-transformer/src/resources/amplify-dynamodb-table/amplify-table-manager-lambda/amplify-table-manager-handler.ts

```diff
@@ -942,6 +967,80 @@ const isKeySchemaModified = (currentSchema: Array<KeySchemaElement>, endSchema:
   return false;
 };
 
+/**
+ * Util function to check if the time to live is modified between old and new table definitions
+ * @param oldTtl TTL config from old table properties
+ * @param endTtl TTL config from input table properties
+ * @returns boolean indicaes the change of TTL
+ */
+export const isTtlModified = (
+  oldTtl: CustomDDB.TimeToLiveSpecificationProperty | undefined,
+  endTtl: CustomDDB.TimeToLiveSpecificationProperty | undefined,
+): boolean => {
+  if (oldTtl === undefined && endTtl === undefined) {
+    return false;
+  }
+  if (oldTtl === undefined || endTtl === undefined) {
+    return true;
+  }
+  return oldTtl.enabled !== endTtl.enabled || oldTtl.attributeName !== endTtl.attributeName;
+};
+
+/**
+ * Get time to live specification for the given table
+ * This API call has a limit rate. The call is staggered randomly within 10s and called with exponential retries
+ * @param tableName table name
+ * @returns time to live specification object
+ */
+const getTtlStatus = async (tableName: string): Promise<DescribeTimeToLiveCommandOutput> => {
+  const initialDelay = Math.floor(Math.random() * 10 * 1000); // between 0 to 10s
```

Comment:
Why would we implement an initial delay? The usual way of handling this would be to start our exponential retries if and when we hit a throttling limit. It seems counterintuitive, not to mention punitive, to introduce a 10 second delay for a 1-table update.

Reply:
I've finally use the combination of `maxRetry=5` and `delay=5s`. Some datapoints are as follows.
| max retry | initial delay (s) | total deployment time (s)|
|---|---|---|
|5|10|407.1|
|5|0|383.82|
|5|5|382.45|